python/lsst/drp/tasks/assemble_cell_coadd.py

```diff
@@ -340,30 +345,49 @@ def run(self, inputWarps, skyInfo, **kwargs):
                     )
                     continue
 
-                stacker.add_masked_image(mi, weight=weight)
-
-                coadd_inputs = coadd_inputs_gc[cellInfo.index]
-                self.input_recorder.addVisitToCoadd(coadd_inputs, warp[bbox], weight)
-                if True:
-                    ccd_table = (
-                        warp.getInfo()
-                        .getCoaddInputs()
-                        .ccds.subsetContaining(cell_centers_sky[cellInfo.index])
-                    )
-                    assert len(ccd_table) > 0, "No CCD from a warp found within a cell."
-                    assert len(ccd_table) == 1, "More than one CCD from a warp found within a cell."
-                    ccd_row = ccd_table[0]
-                else:
-                    for ccd_row in warp.getInfo().getCoaddInputs().ccds:
-                        if ccd_row.contains(cell_centers_sky[cellInfo.index]):
-                            break
+                ccd_table = (
+                    warp.getInfo().getCoaddInputs().ccds.subsetContaining(cell_centers_sky[cellInfo.index])
+                )
+                assert len(ccd_table) > 0, "No CCD from a warp found within a cell."
+                assert len(ccd_table) == 1, "More than one CCD from a warp found within a cell."
+                ccd_row = ccd_table[0]
 
                 observation_identifier = ObservationIdentifiers.from_data_id(
                     warpRef.dataId,
                     backup_detector=ccd_row["ccd"],
                 )
                 observation_identifiers_gc[cellInfo.index].append(observation_identifier)
 
+                stacker = gc[cellInfo.index]
+                stacker.add_masked_image(mi, weight=weight)
+
+                calexp_point = ccd_row.getWcs().skyToPixel(cell_centers_sky[cellInfo.index])
+                undistorted_psf_im = ccd_row.getPsf().computeImage(calexp_point)
+                # Adjust the bounding box of undistorted_psf_im to be centered
+                # at the closest integer location to the requested location.
+                undistorted_psf_im.setXY0(
+                    geom.Box2I.makeCenteredBox(calexp_point, undistorted_psf_im.getDimensions()).getMin()
+                )
+
+                # Convert the PSF image from Image to MaskedImage.
+                undistorted_psf_maskedImage = afwImage.MaskedImageD(image=undistorted_psf_im)
+                # TODO: In DM-43585, use the variance plane value from noise.
```

Comment:
I don't think it matters; the stacker had better not be using this variance for anything other than the output variance (we're passing it an explicit weight below), and we're throwing the output variance away.

Reply:
I have to double check, but we need that variance because its inverse is used as the weight.